python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -386,56 +488,298 @@ def applyBHMicrolens(self, params, expmjd_in):
+        # just in case the user wants to override the light curve cache
+        # file by hand before generating the catalog
+        if len(params) == 0:
+            return numpy.array([[],[],[],[],[],[]])
+
+        if not hasattr(self, 'photParams'):
+            raise RuntimeError("To apply MLT dwarf flaring, your "
+                               "InstanceCatalog must have a member variable "
+                               "photParams which is an instantiation of the "
+                               "class PhotometricParameters, which can be "
+                               "imported from lsst.sims.photUtils. "
+                               "This is so that your InstanceCatalog has "
+                               "knowledge of the effective area of the LSST "
+                               "mirror.")
+
+        if _MLT_LC_CACHE is None or _MLT_LC_CACHE_NAME != self._mlt_lc_file:
+            if not os.path.exists(self._mlt_lc_file):
+                catutils_scripts = os.path.join(getPackageDir('sims_catUtils'), 'support_scripts')
+                raise RuntimeError("The MLT flaring light curve file:\n"
+                                    + "\n%s\n" % self._mlt_lc_file
+                                    + "\ndoes not exist."
+                                    +"\n\n"
+                                    + "Go into %s " % catutils_scripts
+                                    + "and run get_mdwarf_flares.sh "
+                                    + "to get the data")
+
+            _MLT_LC_CACHE = numpy.load(self._mlt_lc_file)
+            _MLT_LC_CACHE_NAME = self._mlt_lc_file
+
+        if not hasattr(self, '_mlt_dust_lookup'):
+            # Construct a look-up table to determine the factor
+            # by which to multiply the flares' flux to account for
+            # dust as a function of E(B-V).  Recall that we are
+            # modeling all MLT flares as 9000K blackbodies.
+
+            if not hasattr(self, 'lsstBandpassDict'):
+                raise RuntimeError('You are asking for MLT dwarf flaring '
+                                   'magnitudes in a catalog that has not '
+                                   'defined lsstBandpassDict.  The MLT '
+                                   'flaring magnitudes model does not know '
+                                   'how to apply dust extinction to the '
+                                   'flares without the member variable '
+                                   'lsstBandpassDict being defined.')
+
+            ebv_grid = numpy.arange(0.0, 7.01, 0.01)
+            bb_wavelen = numpy.arange(200.0, 1500.0, 0.1)
+            hc_over_k = 1.4387e7  # nm*K
+            temp = 9000.0  # black body temperature in Kelvin
+            exp_arg = hc_over_k/(temp*bb_wavelen)
+            exp_term = 1.0/(numpy.exp(exp_arg) - 1.0)
+            ln_exp_term = numpy.log(exp_term)
+
+            log_bb_flambda = -5.0*numpy.log(bb_wavelen) + ln_exp_term
```

Comment:
Yes, but because I only care about the ratio of flux between the cases where there is and is not dust extinction, I am discarding any normalizing factors.